src/libawkward/array/RecordArray.cpp

```diff
@@ -376,9 +376,6 @@ namespace awkward {
   }
 
   const std::shared_ptr<Content> RecordArray::flatten(int64_t axis) const {
-    if (axis != 0) {
-      throw std::runtime_error("FIXME: RecordArray::flatten(axis != 0)");
-    }
     std::vector<std::shared_ptr<Content>> contents;
     for (auto content : contents_) {
       contents.push_back(content.get()->flatten(axis));
```

Comment:
This can only be right if the `axis` doesn't flatten the structure immediately within the RecordArray. That is, if each `content` of the RecordArray does not change its length (because the flattening happens more deeply than this level), all is well. Otherwise, it would break the alignment that makes RecordArray meaningful.

Reply:
addressed in #102